tests/Makefile

```diff
@@ -0,0 +1,53 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the MIT License. See the LICENSE accompanying this file
+# for the specific language governing permissions and limitations under
+# the License.
+#
+# This file provides a common interface to execute the various tests.
+
+TESTS=eth0-basic secondary-basic secondary-more-ips
+SCRIPTS=ec2ifup ec2dhcp.sh ec2ifdown
+
+define do-tests
+ $(foreach t,${SCRIPTS},
+   $(call do-test,${1},${t})
+  )
+endef
+
+define do-test
+  PARAMS=./params-${1}.sh ./test-${2} 2> test-output/${2}.out
+endef
+
+define compare-outputs
+  $(foreach t,${SCRIPTS},
+    $(call compare-output,${1},${t})
+  )
+endef
+
+define compare-output
+  diff -u tdata/${1}/${2}.out test-output/${2}.out
+endef
+
+%:
+	mkdir -p test-output
+	$(call do-tests,$@)
```

Comment:
I think that if you do the content of do-test here, and have a bit more pattern matching fun, you can end up with supporting parallel test runs.

I've used patterns like this in the past to do that:

```
THING_TEST = foo/test/run-thing \
  foo/test/run-thing2 \
  foo/test/run-thing3

check:  $(THING_TEST:%=%-check)
$(THING_TEST:%=%-check) : %-check: %
        $(call Q, RUN-TEST ,$(VALGRIND) $<, $<)
```

Reply:
I've cleaned up the parallelism a bit in the latest update. The real issue was that the scratch space used by the tests was shared between tests, meaning they'd scribble all over each others work. Fixed that by moving to a subdirectory, and tests can be run in parallel.